app/models/enhancer.rb

```diff
@@ -0,0 +1,17 @@
+class Enhancer
+  attr_accessor :enhanced_query
+
+  # accepts all params as each enhancer may require different data
+  def initialize(params)
+    @enhanced_query = {}
+    @enhanced_query[:raw_query] = params
+    @enhanced_query[:q] = params[:q] if params[:q]
+    patterns(params) if params[:q]
+  end
```

Comment:
If I understand what you're setting up here, enhancers which are different than regex matches would get a different method here, and a different subclass? This would then anticipate things like the hints system, or journal name matching, or timely-news detection?

Reply:
Yup